mmv1/third_party/terraform/services/cloudquotas/resource_cloud_quotas_quota_preference_test.go

```diff
@@ -0,0 +1,82 @@
+package cloudquotas_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"project":       envvar.GetTestProjectFromEnv(),
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_basic(context),
+			},
+			{
+				ResourceName:            "google_cloud_quotas_quota_preference.preference",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parent", "quota_preference_id", "allow_missing", "validate_only", "ignore_safety_checks"},
+			},
+			{
+				Config: testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_update(context),
+			},
+			{
+				ResourceName:            "google_cloud_quotas_quota_preference.preference",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parent", "quota_preference_id", "allow_missing", "validate_only", "ignore_safety_checks"},
+			},
+		},
+	})
+}
+
+func testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_cloud_quotas_quota_preference" "preference" {
+  name          = "projects/%{project}/locations/global/quotaPreferences/tf-test-compute_googleapis_com-CPUS-per-project_us-central2%{random_suffix}"
+  quota_config  {
+    preferred_value = 210
+  }
+  dimensions = {
```

Comment:
Ideally, one of these test configs should be minimal, with the least amount of fields provided. That helps to verify that optional fields are in fact optional.

Then, we should be updating all fields that can be mutated (for example, the schema defines `service` as being optional).

Reply:
Done, I added more tests to cover scenarios where mutable fields are modified. . 
I have a question regarding the design:  Our API accepts the `allow-missing` flag for `update`. When enabled, the `update` method is equivalent to `create`. If we take out the flag, certain fields (like `service`, `quotaId` and `dimensions`) become immutable. When the user wants to update those fields, they will have to create a new preference. I'd like your suggestion whether we want to support such feature in TF?

Reply:
Using `allow-missing` could modify the `parent``and `name` fields, which are used to generate the resource ID. This led to provider import errors, so I removed the allow-missing flag.

Reply:
Yea, I think that was the correct outcome here in terms of ergonomics as well. Terraform expects more explicit CRUD actions, so an upsert typically does not fit as well.